docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,576 @@
+  resources: [clusterextensions/finalizers]
+  verbs: [update]
+  resourceNames: [<cluster-extension-name>]
+```
+
+CRUD for Custom Resource Definitions
+
+```yaml
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [create, list, watch]
+
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<crd name 1>, ..., <crd name n>]
+```
+
+CRUD for Cluster Roles
+
+```yaml
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: [create, list, watch]
+
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<generated cluster role 1>, ..., <generated cluster role n>, <manifest cluster role name 1>, ..., <manifest cluster role name n>]
+```
+
+CRUD for Cluster Role Bindings
+
+```yaml
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: [create, list, watch]
+
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<generated cluster role 1>, ..., <generated cluster role n>, <manifest cluster role binding name 1>, ..., <manifest cluster role binding name n>]
+```
+
+CRUD for managing deployments
+
+```yaml
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [create, list, watch]
+
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [get, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+```
+
+CRUD for service accounts used by the deployment
+ 
+```yaml
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [create, list, watch]
+
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [get, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+```
+
+### Creation of ClusterRoleBinding using the cluster-admin ClusterRole in non-production environments
+
+Below is an example ClusterRoleBinding using the cluster-admin ClusterRole for non-production environments.
+
+```yaml
+# Create ClusterRole
+kubectl apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: my-cluster-extension-installer-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: my-cluster-extension-service-account
+  namespace: my-cluster-extension-namespace
+EOF
+```
+
+Use the below on a Kind cluster to assign cluster-admin privileges to your cluster extension
+
+```sh
+kubectl create clusterrolebinding my-cluster-extension-installer-role-binding \
+  --clusterrole=cluster-admin \
+  --serviceaccount=my-cluster-extension-namespace:my-cluster-installer-service-account
+```
+
+### Example ClusterExtension with RBAC
+This is an example of the argocd installer with the necessary RBAC to deploy the ArgoCD ClusterExtension.
+
+```yaml
```

Comment:
I still think these long examples should be placed into an expandable block section to match the rest of our documentation as seen here: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks

Reply:
i believe this is fixed, pls confirm

Reply:
renders correctly now, thanks for your help